seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumComJSRunner.scala

```diff
@@ -0,0 +1,107 @@
+package org.scalajs.jsenv.selenium
+
+import org.scalajs.core.tools.io.{MemVirtualJSFile, VirtualJSFile}
+import org.scalajs.core.tools.jsdep.ResolvedJSDependency
+import org.scalajs.jsenv.ComJSRunner
+
+import java.util.concurrent.TimeoutException
+
+import scala.annotation.tailrec
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+
+class SeleniumComJSRunner(browserProvider: SeleniumBrowser,
+    libs: Seq[ResolvedJSDependency], code: VirtualJSFile, keepAlive: Boolean)
+    extends SeleniumAsyncJSRunner(browserProvider, libs, code, keepAlive)
+    with ComJSRunner {
+
+  protected def envName: String =
+    "SeleniumComJSRunner on " + browserProvider.name
+
+  def send(msg: String): Unit = {
+    awaitForBrowser()
+    browser.getWebDriver.executeScript(
+        "this.scalajsSeleniumComJSRunnerChannel.recvMessage(arguments[0])", msg)
+    browser.processConsoleLogs(console)
+  }
+
+  def receive(timeout: Duration): String = {
+    awaitForBrowser(timeout)
+    @tailrec def loop(): String = {
+      val script = "return this.scalajsSeleniumComJSRunnerChannel.popOutMsg();"
+      browser.getWebDriver.executeScript(script) match {
+        case null =>
+          loop()
+
+        case msg: String =>
+          browser.processConsoleLogs(console)
+          msg
+
+        case obj =>
+          browser.processConsoleLogs(console)
+          throw new IllegalStateException(
+              "Receive ill formed message of type " + obj.getClass +
+              " with value: " + obj)
```

Comment:
Hey this looks like it could be reused in `FirefoxDriver.newConsoleLogsIterator`.